### DISCUSSION TITLE
Help with RLS for tables and storage

### DISCUSSION THREAD

**Discussion by lathinharoon** [supabase/supabase#2819] [Questions] [answered]
Hey, I am fairly new to supabase but i have been enjoying it so far building my first social media app with flutter and supabase. At the moment i am trying to implement RLS for my tables and for the storage but it does not seem to work.

For storage, I have two buckets: profiles and videos. In profiles bucket, user can upload an profile pic with their uid as the parent folder. the user can only insert a file inside there folder.

![image](https://user-images.githubusercontent.com/68173945/129277101-3acd3d18-4def-4d62-905f-472eee62f812.png)

![image](https://user-images.githubusercontent.com/68173945/129277214-0b9b97bd-7eb0-44be-be72-87dfe0e920e8.png)

In flutter, when i try to upload the image:
`
supabase.storage.from('profiles').upload('uid/profilepic1.jpg', file);
`

I get the error:
![image](https://user-images.githubusercontent.com/68173945/129277642-22424380-05a1-4bf4-a525-b76491fa3e12.png)

Same thing happens when I try creating a new row on a table that has RLS enabled and the query satisfies all the conditions. The error says: error (new row violates row-level security policy for table "users")

Does anyone why this is happening? Any help would be greatly appreciated.

Thank you.

**Comment by dshukertjr** [comment, marked as answer]
Hi @lathinharoon !

I tested it out right now with the policy you have and I was able to upload an image successfully!

This might be a stupid question, but you are not literally running this code
```dart
supabase.storage.from('profiles').upload('uid/profilepic1.jpg', file);
```

but doing something like this, correct? 
```dart
final uid = supabase.currentUser!.id;
supabase.storage.from('profiles').upload('$uid/profilepic1.jpg', file);
```

Because if you weren't then that should be the cause.

**Reply by lathinharoon** [reply]
Hey @dshukertjr ,

Thanks for the help! 

I was doing
```
upload(String currentUid, File toUpload) {
    supabase.storage.from('profiles').upload('$uid/profilepic1.jpg', toUpload);
}
```

With Your answer, I realised that the issue was `supabase.auth.currentUser!.id` as it is returning null for the current user after signing up. I am using using get_it to get a singleton class. I am not very sure why that is the case. 

For now, email confirmation is disabled. however, The Current User is still null. 

```
  Future<void> signUp({
    required String email,
    required String password,
  }) async {
    final res = await supabaseClient.auth.signUp(email, password);
    if (res.error != null)
      throw SignUpFailure(message: res.error!.message);
    else if (res.data == null && res.user == null) {
      await prefs.setString('email', email);
      await prefs.setString('pass', password);
      throw EmailVerification();
    } else {
      await prefs.setString(petsistSessionKey, res.data!.persistSessionString);
      print(supabaseClient.auth.currentUser);
    }
  }
```

Thank you.

**Reply by dshukertjr** [reply]
@lathinharoon 

Could you maybe post the entire code from top to bottom if it's not too much? Or if it's on a public repo, you could just drop the URL of it!

When you run 
```dart
Future<void> signUp({
    required String email,
    required String password,
  }) async {
    final res = await supabaseClient.auth.signUp(email, password);
    if (res.error != null)
      throw SignUpFailure(message: res.error!.message);
    else if (res.data == null && res.user == null) {
      await prefs.setString('email', email);
      await prefs.setString('pass', password);
      throw EmailVerification();
    } else {
      await prefs.setString(petsistSessionKey, res.data!.persistSessionString);
      print(supabaseClient.auth.currentUser);
    }
  }
```

It throws `EmailVerification()` correct?

**Reply by lathinharoon** [reply]
Hi @dshukertjr,

Thanks for your reply. Thankfully, I fixed the problem with my code by migrating to supabase_flutter.
